tests/grid/grid_tools_collect_periodic_faces_parallel_tolerance.cc

```diff
@@ -0,0 +1,88 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/point.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+
+#include <vector>
+
+#include "../tests.h"
+
+template <int dim>
+void
+test(MPI_Comm comm)
+{
+  deallog << "dim = " << dim << std::endl;
+  parallel::distributed::Triangulation<dim> tria(comm);
+
+  std::vector<typename GridTools::PeriodicFacePair<
+    typename parallel::distributed::Triangulation<dim>::cell_iterator>>
+    periodicity_vector;
```

Comment:
Can we use a plain `Triangulation` instead? That way this test would not depend on p4est or MPI.